OODebug/src/debugger/JavaDebugger.cpp

```diff
@@ -250,42 +278,67 @@ Interaction::CommandResult* JavaDebugger::probe(OOVisualization::VStatementItemL
 	auto defaultTypeAndHandler = defaultPlotTypeAndValueHandlerFor(vars);
 	auto observer = std::make_shared<VariableObserver>
 			(VariableObserver{defaultTypeAndHandler.second, vars, observedNode, parsedArgs.first});
-	nodeObservedBy_.insertMulti(observedNode, observer);
-	addBreakpointAt(observedNode);
+	qint64 breakpointId = addBreakpoint(observedNode, BreakpointType::Internal);
 
-	auto overlay = new Visualization::IconOverlay{vItem, Visualization::IconOverlay::itemStyles().get("monitor")};
-	vItem->addOverlay(overlay, MONITOR_OVERLAY_GROUP);
+	auto monitorOverlay = new Visualization::IconOverlay{vItem, Visualization::IconOverlay::itemStyles().get("monitor")};
+	vItem->addOverlay(monitorOverlay, MONITOR_OVERLAY_GROUP);
 
 	auto plotOverlay = new PlotOverlay{vItem, PlotOverlay::itemStyles().get("default"),
 												  defaultTypeAndHandler.first, variableNames};
 	vItem->addOverlay(plotOverlay, PLOT_OVERLAY_GROUP);
+
+	probes_.insertMulti(observedNode, {breakpointId, monitorOverlay, plotOverlay});
+
+	addBreakpointListener(breakpointId, createListenerFor(observer, plotOverlay));
+
+	qDebug() << "Added probe" << breakpointId;
+
 	return new Interaction::CommandResult{};
 }
 
-void JavaDebugger::addBreakpoint(Model::Node* location, BreakpointType type)
+qint64 JavaDebugger::addBreakpoint(Model::Node* location, BreakpointType type)
 {
-	if (BreakpointType::Internal == type)
-		addBreakpointAt(location);
-	else
+	qint64 breakpointId = ++nextBreakpointId_;
```

Comment:
So it does seem like you're just creating the breakpoint ids yourself. Do we really need to be able to create int64 amount of breakpoints?